src/liballoc_jemalloc/lib.rs

```diff
@@ -211,17 +215,28 @@ mod contents {
     #[no_mangle]
     #[linkage = "external"]
     pub unsafe extern fn __rde_realloc_excess(ptr: *mut u8,
-                                              old_size: usize,
+                                              _old_size: usize,
                                               old_align: usize,
                                               new_size: usize,
                                               new_align: usize,
                                               excess: *mut usize,
                                               err: *mut u8) -> *mut u8 {
-        let p = __rde_realloc(ptr, old_size, old_align, new_size, new_align, err);
-        if !p.is_null() {
-            *excess = new_size;
+        if new_align != old_align {
+            ptr::write(err as *mut AllocErr,
+                       AllocErr::Unsupported { details: "can't change alignments" });
+            return 0 as *mut u8
         }
-        return p
+
+        let flags = align_to_flags(new_align);
+        let ptr = rallocx(ptr as *mut c_void, new_size, flags) as *mut u8;
+        let alloc_size = sallocx(ptr as *mut c_void, flags);
```

Comment:
`sallocx` and `*excess =` need to go into a !null branch.

Reply:
!!! indeed, should be fixed now